app/assets/javascripts/applications.js

```diff
@@ -1,2 +1,24 @@
 // Place all the behaviors and hooks related to the matching controller here.
 // All this logic will automatically be available in application.js.
+
+function ajaxUpdateApplicationStatus(form, errorMessage) {
+  if (!window.FormData)
+    return form.submit();
+
+  var xhr = new XMLHttpRequest();
```

Comment:
Tried that first. It looked like turbolinks intercepted the requests and at some point I grow tired of debugging it. Line- and complexity-wise it didn't make that much of a difference anyways